File: npm/lib/init-command.js
const fs = require("fs");
const path = require("path");
const { contentRoot } = require("./content-root");
const { fail } = require("./errors");
const { copyDirectory, ensureNoUnknownArgs } = require("./fs-utils");

function initProject(args) {
  ensureNoUnknownArgs(args, ["--force"]);
  const force = args.includes("--force");
  const source = path.join(contentRoot, "src", "canonical", "project-files", "specs");
  const destination = path.join(process.cwd(), ".specs");

  if (!fs.existsSync(source)) {
    fail(`Bundled canonical project files not found: ${source}`);
  }

  if (fs.existsSync(destination) && !force) {
    fail("File already exists: .specs\nUse --force to overwrite.");
  }

  copyDirectory(source, destination, force);
  console.log("Initialized .specs.");
}

module.exports = { initProject };

File: tools/idd-tool/Commands/InitCommand.cs
internal sealed class InitCommand
{
    public int Run(IReadOnlyList<string> commandArgs)
    {
        EnsureNoUnknownOptions(commandArgs, "--force");
        var force = commandArgs.Contains("--force", StringComparer.Ordinal);
        var source = new ContentLayout(ContentRootLocator.Find()).IntentRoot;
        var destination = Path.Combine(Directory.GetCurrentDirectory(), ".idd/intent");

        if (!Directory.Exists(source))
        {
            return Fail($"Bundled canonical project files not found: {source}");
        }

        if (Directory.Exists(destination) && !force)
        {
            return Fail("File already exists: .idd/intent" + Environment.NewLine + "Use --force to overwrite.");
        }

        DirectoryCopier.Copy(source, destination, force);
        Console.WriteLine("Initialized .idd/intent.");
        return 0;
    }

    private static void EnsureNoUnknownOptions(IReadOnlyList<string> commandArgs, params string[] known)
    {
        foreach (var arg in commandArgs.Where(arg => arg.StartsWith("--", StringComparison.Ordinal)))
        {
            if (!known.Contains(arg, StringComparer.Ordinal))
            {
                throw new ToolException($"Unknown option: {arg}");
            }
        }
    }

    private static int Fail(string message)
    {
        Console.Error.WriteLine(message);
        return 1;
    }
}
